src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -97,64 +128,40 @@ private void initializeTranscript(Entity lectureEntity) throws IOException, Serv
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long lectureId = Long.parseLong(request.getParameter(LectureUtil.ID));
+    if (request.getParameter(PARAM_ID) == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, ERROR_MISSING_ID);
+      return;
+    }
+    long lectureId = Long.parseLong(request.getParameter(PARAM_ID));
```

Comment:
Why are you using "PARAM_ID" instead of LectureUtil.ID?

Reply:
We try to use PARAMs defined within the servlet that uses them.  This is because they're defined in each corresponding javascript file.  If we change LectureUtil.ID, it would be hard to know what needs to also change on the javascript side

Reply:
Are you saying LectureUtil.ID is not the same as PARAM_ID?

Reply:
They have the same value

Reply:
But they have different meanings.  I think LectureUtil.ID was added so we didn't need to define the same PARAM in multiple servlets, but this is a bad idea because we declare them separately in the javascript.  

What we do everywhere else is define the PARAMs that the servlet's requests will be using within the servlet itself.  There may be some duplicate values, but this way if one needs to change, it doesn't break all the other servlets that were using it.

Reply:
Ok sounds good. Thank you :)